flake.nix

```diff
@@ -44,6 +44,21 @@
     in flake // {
       packages.default = flake.packages."pact:exe:pact";
 
+      packages.docs = pkgs.stdenv.mkDerivation {
+        name = "pact-docs";
+        src = ./docs;
+        buildInputs = [
+          (pkgs.python3.withPackages (ps: [ps.sphinx ps.sphinx_rtd_theme]))
+          pkgs.pandoc
+          pkgs.perl
+        ];
+        buildPhase = ''
+          source ./work.sh
+          mkdir $out
+          cp -r _build/* $out
+        '';
+      };
```

Comment:
Presumably, this is calling `./work.sh` because the nix-shell is now redundant. This creates a conflict that we need to figure out: do we make the nix flake the canonical worker, and modify `build.sh` to drop into the flake shell?

Reply:
Good point! `build.sh` could either run `nix build .#docs` or `nix develop .#docs --command "./work.sh"`.

I suggest going with the `nix develop` alternative, because if someone is running `build.sh` and not `nix build .#docs`, chances are they're not interested in dealing with Nixisms, in which case, they probably don't want to worry about how Nix flakes ignore files that are not tracked by git. 

Either way, it's probably also a good idea to explicitly add the `--experimental-features "nix-command flakes"` argument to the `nix` call too.